### ISSUE TITLE
Exercise 3 Level 6

### ISSUE THREAD

**Issue by eesean** [nus-cs2030/2223-s2#93]
Hi all,

Would like to clarify what is meant by the following:

Within SolidCuboid, delegate the computation of mass to a SolidImpl property in the class.

Does this mean we have to call super(density) since SolidImpl has the density attribute? Or do we just use SolidCuboid's density attribute in the calculation?

**Comment by trwstin** [comment]
In simpler terms, you first have to define a SolidImpl attribute in your SolidCuboid class (like how you did with height, density, etc).

Then, make use of this attribute to directly return SolidImpl's mass method within SolidCuboid's mass method.

**Comment by EpicSpirito** [comment]
You can refer to Lecture 5 Slides 5-7 and see how ImList delegates the computation of various methods to an ArrayList property in the class.

**Comment by eesean** [comment]
In that case, would a new constructor be required for delegation? The test cases only allow 4 arguments for creating SolidCuboid objects (height, width, length, density).

**Comment by geoffong11** [comment]
> In that case, would a new constructor be required for delegation? The test cases only allow 4 arguments for creating SolidCuboid objects (height, width, length, density).

A new constructor will not be required. You can create a SolidImpl with the information given within the SolidCuboid class.

**Comment by e0959170** [comment]
u cannot use super because SolidCuboid is not supposed to inherit from SolidImpl! what you can do to utilise the mass function from SolidImpl is to inititate a SolidImpl object and return its mass in ur SolidCuboid mass method

**Comment by eesean** [comment]
Yep I got it now, thanks for the insights!

**Comment by geraldckj** [comment]
this thread was very helpful for anyone struggling with the the question phrasing. Thanks!
